src/main/java/database/SongDsData.java

```diff
@@ -10,14 +10,6 @@
 public class SongDsData {
     private final Song song;
 
-    /**
-     * Constructor to be called from use case code to communicate with database
-     * @param song song object to encapsulate
-     */
-    public SongDsData(Song song){
```

Comment:
Wasn't this constructor necessary for uploading a song?  Or do you do it differently somehow

Reply:
It uses the other one -- and all the save/read stuff is in SongLibrary so it'd be counterintuitive to create a Song object inside database only to put it inside a SongDsData lol. Also Intellij prompted that it was never used 🤷